redwood-treehouse-guest/src/commonMain/kotlin/app/cash/redwood/treehouse/ZiplineFrameClock.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.treehouse
+
+import androidx.compose.runtime.BroadcastFrameClock
+import androidx.compose.runtime.MonotonicFrameClock
+
+/** A guest frame clock that asks the host for frames. */
+internal class ZiplineFrameClock(
```

Comment:
I think you an delete this whole class and replace it with
```kotlin
val clock = BroadcastFrameClock { host.requestFrame() }
```
at the call site. The lambda is invoked when the number of awaiters (i.e., people who have called `withFrameNanos`) goes from 0 to 1.